src/project.rs

```diff
@@ -213,3 +226,35 @@ pub fn collect_source(
     }
     Ok(())
 }
+
+fn check_app_file_version_matches(
+    root: impl AsRef<Path>,
+    project_config: &PackageConfig,
+) -> Result<(), Error> {
+    let mut app_src_path = root.as_ref().to_path_buf();
+    app_src_path.push("src");
+    app_src_path.push(format!("{}.app.src", &project_config.name));
+
+    let contents = std::fs::read_to_string(&app_src_path)
+        .map_err(|err| Error::FileIO {
+            action: FileIOAction::Read,
+            kind: FileKind::File,
+            err: Some(err.to_string()),
+            path: app_src_path.clone(),
+        })?
+        // Remove all new lines so we can regex easily across the content
+        .replace("\n", "");
+
+    // Search for the vsn line & extract the text in double quotes
+    let re = Regex::new("\\{ *vsn *, *\"([^\"]*)\" *\\}").unwrap();
+    let caps = re.captures(&contents).unwrap();
+
+    if caps.get(1).unwrap().as_str() == project_config.version {
```

Comment:
Let's make this function return OK if it cannot determine the version number, it is a best-attempt. We don't want to ever call `unwrap` and crash the program.

Reply:
Done - yeah, I still don't have much head wrapped around error handling in Rust. I guess I should try to remember that plain `unwrap` is going to panic and shouldn't be used. I'll try to keep it in mind.

I've changed the whole thing to chained calls. I hope it isn't going too far. I suspect the function could be refactored to be more testable if needed but it seems to work.

Reply:
Looks great! Really digging your functional style here.